### PR TITLE
Add api server isolation and db seeding

### PR DESCRIPTION
**Pull request by tuanchien**
- Isolate local api server in its own docker container and own scripts
- Create framework for seeding api db data from scratch.
- Print output from observatory platform command failures.

### CODE REVIEW

**Review by jdddog — CHANGES_REQUESTED** [review]
Hey Tuan, thanks for your PR, it looks pretty good. See below for comments.

Now that the apiserver has been separated, [entrypoint-airflow.sh.jinja2](https://github.com/The-Academic-Observatory/observatory-platform/blob/INF-406/INF-405/api_server_isolate_and_seed/observatory-platform/observatory/platform/docker/entrypoint-airflow.sh.jinja2) should be reverted back to how it was.

Could you also reformat the files with Black?

**Comment by tuanchien** [comment]
liccheck needs to be turned off for markupsafe requirement for now.
Due to an api change, it seems like we need a more recent markupsafe version (>=2) but the current airflow version wants to pin it to 1.1.x.  Airflow 2.3.x will use markupsafe >=2 though.

File: observatory-platform/observatory/platform/docker/docker-compose.observatory.yml.jinja2 [hunk 1]
```diff
@@ -307,12 +313,32 @@ services:
...
 
+  seed_db:
+    build:
+      context: .
+      dockerfile: Dockerfile.seed_db
+      args:
+        - HOST_USER_ID=${HOST_USER_ID}
+    environment:
+      - API_URI={{api_uri}}
+    volumes: *volumes
```
Comment (jdddog) [inline observatory-platform/observatory/platform/docker/docker-compose.observatory.yml.jinja2:336]: Might as well mount all volumes as it makes it tidier and we need them for installing from editable.

File: observatory-platform/observatory/platform/docker/docker-compose.observatory.yml.jinja2 [hunk 2]
```diff
@@ -307,12 +313,32 @@ services:
...
   {%- endif %}
 
+  seed_db:
+    build:
+      context: .
+      dockerfile: Dockerfile.seed_db
+      args:
+        - HOST_USER_ID=${HOST_USER_ID}
+    environment:
+      - API_URI={{api_uri}}
```
Comment (jdddog) [inline observatory-platform/observatory/platform/docker/docker-compose.observatory.yml.jinja2:335]: Cleaner to just set `API_URI` environment variable by itself. The api_uri variable is hard coded up the top in the local environment and set from an environment variable in the remote environment.

File: observatory-platform/observatory/platform/docker/seed_db.sh.jinja2 [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+{% endif %}
+{% endfor %}
+
+{% for package in config.python_packages %}
+{% if package.name != 'observatory-platform' and package.name != 'observatory-api' %}
+# Call the command for seeding the database
+echo "Seeding {{ package.name }}"
+{{ package.name }}-seed
+{% endif %}
+{% endfor %}
```
Comment (jdddog) [inline observatory-platform/observatory/platform/docker/seed_db.sh.jinja2:27]: For all workflows packages call the seeding command exposed from setup.cfg.

File: observatory-platform/observatory/platform/docker/Dockerfile.package_install.jinja2 [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Author: Tuan Chien
+
+USER ${INSTALL_USER}
+{% if package.type == 'editable' %}
+
+{% if install_deps %}
```
Comment (jdddog) [inline observatory-platform/observatory/platform/docker/Dockerfile.package_install.jinja2:20]: Added a variable that lets you stop dependencies being installed to make installation of the package quicker.

File: observatory-platform/observatory/platform/docker/docker-compose.observatory.yml.jinja2 [hunk 5]
```diff
@@ -33,13 +33,15 @@ version: '3.8'
 {%- set redis_hostname="redis" -%}
 {%- set api_server_db="observatory" -%}
 {%- set api_server_port="5002" -%}
+{%- set api_uri="http://apiserver:5002" -%}
```
Comment (jdddog) [inline observatory-platform/observatory/platform/docker/docker-compose.observatory.yml.jinja2:36]: Can just be hard coded in local environment.

File: observatory-platform/observatory/platform/docker/docker-compose.observatory.yml.jinja2 [hunk 6]
```diff
@@ -33,13 +33,15 @@ version: '3.8'
...
 {%- else -%}
 
 {%- set host_data_path="/opt/observatory/data" -%}
 {%- set host_logs_path="/opt/airflow/logs" -%}
 {%- set google_application_credentials="/opt/observatory/google_application_credentials.json" -%}
 {%- set postgres_hostname="${POSTGRES_HOSTNAME}" -%}
 {%- set redis_hostname="${REDIS_HOSTNAME}" -%}
+{%- set api_uri="${API_URI}" -%}
```
Comment (jdddog) [inline observatory-platform/observatory/platform/docker/docker-compose.observatory.yml.jinja2:44]: In remote environment get from environment variable.

**Review by jdddog — APPROVED** [review]
Assuming that we have the tests passing it looks good to merge Tuan

**Pull request merged**